src/main/java/com/novocode/junit/EventDispatcher.java

```diff
@@ -148,12 +148,6 @@ void logTo(RichLogger logger) {
     });
   }
 
-  private void postIfFirst(AbstractEvent e)
-  {
-    e.logTo(logger);
-    if(reported.add(e.fullyQualifiedName())) handler.handle(e);
```

Comment:
`reported` now goes unused.

Reply:
`AbstractEvent` should probably expose `Description`, and `reported` should use the description's identity to track whether the event has been posted or not.